sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -211,50 +216,44 @@ public void AddTag(string name, object value)
                 }
             }
 
-            private List<ActivityLink>? GetActivitySourceLinkCollection()
+            private IEnumerable<Activity> GetDiagnosticSourceLinkCollection()
             {
                 if (_links == null)
                 {
-                    return null;
+                    return Array.Empty<Activity>();
                 }
 
-                var linkCollection = new List<ActivityLink>();
+                var linkCollection = new List<Activity>();
 
-                foreach (var activity in _links)
+                foreach (var link in _links)
                 {
-                    ActivityTagsCollection linkTagsCollection = new();
-                    foreach (var tag in activity.Tags)
+                    var activity = new Activity("LinkedActivity");
+                    if (link.Context != default) // TODO
                     {
-                        linkTagsCollection.Add(tag.Key, tag.Value!);
+                        string flags = (link.Context.TraceFlags == ActivityTraceFlags.None) ? "00" : "01";
```

Comment:
Are there other scenarios where we need to transform `TraceFlags` into a string? Should we create a helper to make this reusable and easier to interpret?

Reply:
yes, there might be other use cases for creating string from activity context, will create helper method for it.